.github/workflows/pull-request-ci.yaml

```diff
@@ -0,0 +1,307 @@
+    - uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: cache.tools
+      uses: actions/cache@v2
+      with:
+        path: ${{ github.workspace }}/tools
+        key: ${{ runner.os }}-tools
+    - run: make tools-install
+    - run: make kubecf-bundle
+      env:
+        TARGET_FILE: ${{ github.workspace }}/kubecf-bundle.tgz
+    - run: |
+        mkdir bundle
+        tar xf kubecf-bundle.tgz -C bundle
+      working-directory: ${{ github.workspace }}
+    - name: Upload bundle
+      uses: actions/upload-artifact@v2
+      with:
+        name: kubecf-bundle.tgz
+        path: ${{ github.workspace }}/bundle
+
+  tests:
+    name: tests
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental }}
+    needs: build
+    env:
+      # For KubeCF
+      PINNED_TOOLS: true
+      TOOLS_DIR: ${{ github.workspace }}/tools
+      FEATURE_AUTOSCALER: "true"
+      # For catapult
+      # Note that credentials-related parts are in the individual steps
+      BACKEND: gke
+      DOWNLOAD_CATAPULT_DEPS: "false"
+
+    defaults:
+      run:
+        working-directory: kubecf
+
+    strategy:
+      matrix:
+        include:
+        - backend: diego
+          experimental: false
+        - backend: eirini
+          experimental: true
+
+    steps:
+    - name: Checkout KubeCF
+      uses: actions/checkout@v2
+      with:
+        path: kubecf
+        submodules: recursive
+    - name: Checkout catapult
+      uses: actions/checkout@v2
+      with:
+        repository: SUSE/catapult
+        path: catapult
+    - uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: cache.tools
+      uses: actions/cache@v2
+      with:
+        path: ${{ github.workspace }}/tools
+        key: ${{ runner.os }}-tools
+    - run: make tools-install
+
+    - id: download-bundle
+      name: Download chart bundle
+      uses: actions/download-artifact@v2
+      with:
+        name: kubecf-bundle.tgz
+        # The artifact upload/download creates an extra directory
+        path: ${{ github.workspace }}/kubecf-bundle
+
+    - name: Download gomplate
+      run: |
+        if test -x "${TOOLS_DIR}/gomplate" ; then
+          echo "gomplate already exists; skipping."
+          exit 0
+        fi
+        mkdir -p "${TOOLS_DIR}"
+        wget -O "${TOOLS_DIR}/gomplate" \
+          https://github.com/hairyhenderson/gomplate/releases/latest/download/gomplate_linux-amd64
+        chmod a+x "${TOOLS_DIR}/gomplate"
+
+    - name: Start SSH agent
+      run: |
+        set -o errexit -o pipefail -o nounset
+        eval "$(ssh-agent -s)"
+        ssh-keygen -t rsa -b 4096 -N '' -C "KubeCF CI #${{ github.run_id }}" -f ssh-key
+        ssh-add ssh-key
+        echo "::add-mask::$(cat ssh-key.pub)"
+        rm -f ssh-key ssh-key.pub
+        echo "::set-env name=SSH_AUTH_SOCK::${SSH_AUTH_SOCK}"
+        echo "::set-env name=SSH_AGENT_PID::${SSH_AGENT_PID}"
+
+    - name: set GKE_CRED_JSON
```

Comment:
Please use the vault action [1] for these secrets with our vault deployment [2].
Please read the guide here on how to use a personal access token to login [3].
The `VAULT_TOKEN` is already available to kubecf.

[1] https://github.com/marketplace/actions/vault-secrets
[2] http://volt.cap.explore.suse.dev/
[3] https://www.vaultproject.io/docs/auth/github#github-auth-method

Reply:
https://github.com/cloudfoundry-incubator/kubecf/issues/1219